### PR TITLE
Add support to authenticate to GKE clusters using authentication plug-ins or tokens.

### PR DESCRIPTION
**Pull request by aosoriodw**
## Description
Add support for requesting Kubeconfig with and without and authentication provider when creating a GKE cluster.

## Blast Radius
None since tests use a pinned version

### Documentation
- [x] I have updated user facing documentation.
- [ ] I have updated all Runbooks affected by this change.
- [ ]  I updated `DEVELOPING.md` with any dev tricks I used to work on this code efficiently.
- [ ] My changes do not have any impact on documentation.

### Monitoring
- [ ] I have verified that any changes to alerts work as expected.
- [ ] My changes affect monitoring rules and/or dashboards, and I made sure they didn't break.
- [x] My changes do not have any impact on monitoring.

### External dependencies
- [ ] I have validated that dependencies impacted by this change work as expected.
- [x] My changes do not impact external dependencies.

### Rosie the Robot checklists
- [ ] My changes affect one or more Rosie checklists.
- [x] My changes do not have any impact on Rosie's checklists.

## Testing
- [x] I have validated that my changes work as expected.
- [ ] My changes do not require testing.

### Testing Strategy
Used test workflows

## Security
- [x] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

## Deployment plan
N/A

### CODE REVIEW

File: .github/workflows/smoke.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ on:
 jobs:
   release_smoke:
     strategy:
+      fail-fast: false
```
Comment (AidanWenzel) [inline .github/workflows/smoke.yaml:13]: That's a good catch - will keep that in mind in future test frameworks.

File: docs/GITHUB_ACTIONS.md [hunk 2]
```diff
@@ -31,6 +31,12 @@ jobs:
         clusters:
          - distribution: GKE
            version: "1.21"
+           gcloudCompatible: "false"
+           useAuthProvider: "false"
+         - distribution: GKE
+           version: "1.21"
```
Comment (AidanWenzel) [inline docs/GITHUB_ACTIONS.md:36]: We'll need to keep this in mind when we do the k8s version jumps

**Review by AidanWenzel — DISMISSED** [review]
👍 :shipit:

**Comment by aosoriodw** [comment]
@AidanWenzel I pushed a couple of changes to fix the build failures. Please take a look again

**Review by AidanWenzel — APPROVED** [review]
Very nice 👍

**Pull request merged**